tests/vector3d.h

```diff
@@ -0,0 +1,15 @@
+#ifndef VECTOR3D_H
```

Comment:
maybe rename this file `native_type_test_impl.h` or similar, to make it clear it is part of a test, not an actual 3d vector data type that we need. Since it is just a test, you can merge the 2 headers in 1, and stick the code in the .cpp in test.cpp

Reply:
Done